### ISSUE TITLE
Inference time reduce

### ISSUE THREAD

**Issue by devendraswamy** [ultralytics/yolov5#7955]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I want to reduce the inference time for my models and if I reduce below values then is it reduce the Inference time with loosing prediction accuracy ? please let me know , Thank you in advance.
max_nms = 30000  # maximum number of boxes into torchvision.ops.nms() 
time_limit = 0.030 * bs  # seconds to quit after

### Additional

please comment if any others methods are available for reduction time reduce for onnx models , Thank you.

**Comment by glenn-jocher** [comment]
@devendraswamy 👋 Hello! Thanks for asking about **inference speed issues**. PyTorch Hub speeds will vary by hardware, software, model, inference settings, etc. Our default example in Colab with a V100 looks like this:

<img width="1462" alt="Screen Shot 2022-05-03 at 10 20 39 AM" src="https://user-images.githubusercontent.com/26833433/166505744-b7fe9b9a-d2b3-4e95-9441-fa540e0804ef.png">


YOLOv5 🚀 can be run on CPU (i.e. `--device cpu`, slow) or GPU if available (i.e. `--device 0`, faster). You can determine your inference device by viewing the YOLOv5 console output:

### detect.py inference
```bash
python detect.py --weights yolov5s.pt --img 640 --conf 0.25 --source data/images/
```

<img width="1039" alt="Screen Shot 2022-05-03 at 2 48 42 PM" src="https://user-images.githubusercontent.com/26833433/166572240-ae598b34-1084-40f0-a19b-bf00fd3f9050.png">


### YOLOv5 PyTorch [Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) inference
```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')

# Images
dir = 'https://ultralytics.com/images/'
imgs = [dir + f for f in ('zidane.jpg', 'bus.jpg')]  # batch of images

# Inference
results = model(imgs)
results.print()  # or .show(), .save()
# Speed: 631.5ms pre-process, 19.2ms inference, 1.6ms NMS per image at shape (2, 3, 640, 640)
```

### Increase Speeds

If you would like to increase your inference speed some options are:

- Use batched inference with [YOLOv5 PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading)
- Reduce `--img-size`, i.e. 1280 -> 640 -> 320
- Reduce model size, i.e. YOLOv5x -> YOLOv5l -> YOLOv5m -> YOLOv5s -> YOLOv5n
- Use half precision FP16 inference with `python detect.py --half` and `python val.py --half`
- Use a faster GPUs, i.e.: P100 -> V100 -> A100
- [Export](https://docs.ultralytics.com/yolov5/tutorials/model_export) to ONNX or OpenVINO for up to 3x CPU speedup ([CPU Benchmarks](https://github.com/ultralytics/yolov5/pull/6613))
- [Export](https://docs.ultralytics.com/yolov5/tutorials/model_export) to TensorRT for up to 5x GPU speedup ([GPU Benchmarks](https://github.com/ultralytics/yolov5/pull/6963))
- Use a free GPU backends with up to 16GB of CUDA memory: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

Good luck 🍀 and let us know if you have any other questions!

**Comment by dnth** [comment]
@devendraswamy you can also consider using DeepSparse to significantly speedup the inference speed. I wrote a tutorial on my blog on how I got 180+ FPS inference on a CPU using only 4 cores.

Hope you'll find it useful 👇

https://dicksonneoh.com/portfolio/supercharging_yolov5_180_fps_cpu/

I also wrote a tutorial on OpenVINO 👇

https://dicksonneoh.com/portfolio/how_to_10x_your_od_model_and_deploy_50fps_cpu/

**Comment by devendraswamy** [comment]
Thank you for valuable reply , please tell me can i reduce the max_nms ,time_limit  and if i reduce the max_nms and time_limit  then is it effecting the results of predictions ?

**Comment by glenn-jocher** [comment]
@devendraswamy you can change anything you want. Changes may have natural consequences.
